router/usecase/candidate_routes.go

```diff
@@ -147,7 +147,7 @@ func GetCandidateRoutes(pools []sqsdomain.PoolI, tokenIn sdk.Coin, tokenOutDenom
 // GetCandidateRoutesNew new algorithm for demo purposes.
 // Note: implementation is for demo purposes and is to be further optimized.
 // TODO: spec, unit tests via https://linear.app/osmosis/issue/DATA-250/[candidaterouteopt]-reimplement-and-test-getcandidateroute-algorithm
-func GetCandidateRoutesNew(poolsByDenom map[string][]sqsdomain.PoolI, tokenIn sdk.Coin, tokenOutDenom string, maxRoutes, maxPoolsPerRoute int, logger log.Logger) (sqsdomain.CandidateRoutes, error) {
+func GetCandidateRoutesNew(poolsByDenom map[string][]sqsdomain.PoolI, tokenIn sdk.Coin, tokenOutDenom string, maxRoutes, maxPoolsPerRoute int, minPoolLiquidityCap uint64, logger log.Logger) (sqsdomain.CandidateRoutes, error) {
```

Comment:
Just a  suggestion, instead of passing `maxRoutes, maxPoolsPerRoute int, minPoolLiquidityCap uint64`  as individual parameters encapsulate those and pass as dedicaded config structure:

```
type Config struct {
MaxRoutes int
MaxPoolsPerRoute int
MinPoolLiquidityCap uint64
}
```

That provides several benefits including but not limited reducing cognitive load, documenting each of the parameter in single place, encapsulating validations for the parameters and making consructor more lightweight.

Reply:
Thanks for the suggestion!

This method is still evolving - do you mind if we keep it like this for now, and I will revisit as part of [this task](https://linear.app/osmosis/issue/DATA-250/[candidaterouteopt]-reimplement-and-test-getcandidateroute-algorithm)

The reason why I'm reluctant to apply the change right not is because being explicit about what the parameters are for this function actually help to reason about its dependencies during active development.

Our [router config](https://github.com/osmosis-labs/sqs/blob/48713b07f622e2c377d37598ab77f4b6d7e18ede/config.json#L11-L42) is quite complex. As a result, if we abstract the inputs behind a `Config` name, it will make it more difficult to tell what subset of the main config fields are used here.

So I would like to, first, get to a final implementation, and then think about abstracting it in a more streamlined way in a separate PR as well as cover the function by unit tests